cdap-api/src/main/java/co/cask/cdap/api/lineage/operation/Join.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.operation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents join operation.
+ */
+public class Join extends Operation {
+  private final List<Input> constraints;
+  private final Type joinType;
+  private final List<String> outputs;
+
+  /**
+   * Create an instance of join operation.
+   * @param name the name of the join operation
+   * @param description the description of the join operation
+   * @param constraints the list of inputs participate in the join condition
```

Comment:
so, we will only support join by equality, right? Then both the left and right hand can have multiple keys. For example:
```
... where a.key = b.id && a.source = b.origin
```
So the constraints would be given as a pair of lists of inputs.

Reply:
Added new static inner `Constraint` class which keeps list of Input corresponding to left and right.